iceoryx2-bb/posix/src/read_write_mutex.rs

```diff
@@ -206,25 +207,25 @@ impl ReadWriteMutexBuilder {
 ///
 /// Is returned by [`ReadWriteMutex::read_blocking_lock()`] and [`ReadWriteMutex::read_try_lock()`].
 #[derive(Debug)]
-pub struct MutexReadGuard<'a, 'b, T: Debug> {
-    mutex: &'a ReadWriteMutex<'b, T>,
+pub struct MutexReadGuard<'handle, T: Debug> {
+    handle: &'handle ReadWriteMutexHandle<T>,
 }
 
-unsafe impl<T: Send + Debug> Send for MutexReadGuard<'_, '_, T> {}
-unsafe impl<T: Send + Sync + Debug> Sync for MutexReadGuard<'_, '_, T> {}
+unsafe impl<T: Send + Debug> Send for MutexReadGuard<'_, T> {}
+unsafe impl<T: Send + Sync + Debug> Sync for MutexReadGuard<'_, T> {}
 
-impl<T: Debug> Deref for MutexReadGuard<'_, '_, T> {
+impl<T: Debug> Deref for MutexReadGuard<'_, T> {
     type Target = T;
 
     fn deref(&self) -> &Self::Target {
-        unsafe { (*self.mutex.handle.value.get()).as_ref().unwrap() }
+        unsafe { (*self.handle.value.get()).as_ref().unwrap() }
     }
 }
 
-impl<T: Debug> Drop for MutexReadGuard<'_, '_, T> {
+impl<T: Debug> Drop for MutexReadGuard<'_, T> {
     fn drop(&mut self) {
-        if self.mutex.release().is_err() {
-            fatal_panic!(from self.mutex, "This should never happen! Failed to release read lock.");
+        if ReadWriteMutex::release(self.handle).is_err() {
```

Comment:
Here the implementation is kept in the `ReadWriteMutex` since both `MutexReadGuard` and `MutexWriteGuard` have to call it.